lib/models/ObjectMDLocation.js

```diff
@@ -33,9 +33,20 @@ class ObjectMDLocation {
         return this._data.dataStoreName;
     }
 
+    /**
+     * Update data location with new info
+     *
+     * @param {object} location - single data location info
+     * @param {string} location.key - data backend key
+     * @param {string} location.dataStoreName - type of data store
+     * @param {number} [location.cryptoScheme] - if location data is
+     * encrypted: the encryption scheme version
+     * @param {string} [location.cipheredDataKey] - if location data
+     * is encrypted: the base64-encoded ciphered data key
+     * @return {ObjectMDLocation} return this
+     */
     setDataLocation(location) {
-        this._data.key = location.key;
-        this._data.dataStoreName = location.dataStoreName;
+        this._data = Object.assign(this._data, location);
```

Comment:
@miniscruff I realized that neither the old code nor the change to use `Object.assign` are correct: indeed, we need to *overwrite* any encryption field with the one provided in the new location, i.e. the difference is that if the encryption params do not exist in the `location` parameter, they must be erased from `this._data.` I saw this in the Zenko tests that contain ObjectMDLocation tests, I'm considering backporting them to this 7.10 branch as well.

Reply:
oh ok, that makes sense. Will re-review when it is ready.